data/geany.glade

```diff
@@ -6321,6 +6321,7 @@
               <object class="GtkMenuBar" id="menubar1">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
+                <signal name="deactivate" handler="on_menubar1_deactivate" after="yes" swapped="no"/>
```

Comment:
https://developer.gnome.org/gtk3/stable/GtkMenuShell.html#id-1.3.20.5.10.4 suggests using select/deselect is just fine.  And in practice it does seem to work great.
But agreed the docs are not very clear.